### ISSUE TITLE
Using milestones for minor and major version releases

### ISSUE THREAD

**Issue by anasalkouz** [opensearch-project/OpenSearch#1232]
** Problem Statement **
Currently, we are using labels to determine release version for this issue and pull request. This could explode the list of labels significantly and harder to track the progress for specific release.

** Proposal **
Using Milestones for tracking issues and pull requests for minor and major version releases and keep the use of Labels sort issues by type, priority, severity, functional area, functional group.

![image](https://user-images.githubusercontent.com/25317259/132598882-286ad40a-925b-4e42-905e-05e784b55cac.png)

**Comment by nknize** [comment]
+1 to using the GitHub milestones feature. 

Just a couple initial questions:

* Can anyone add an issue to a milestone?
* Will milestones be time based (e.g., minor version milestone planned every N weeks)?
* How do version labels cause explosion if maintainers are diligent in labeling and communicating with /asking questions of other contributors & community? Maybe combinatorial explosion of labels is a separate issue? 
* Do we open a separate issue to discuss logically organizing source "components" into separate GitHub teams (e.g., analytics, search, distribution) to make notifications easier? And maybe how maintainers join those teams?

**Comment by praveensameneni** [comment]
Milestones can be used to track progress of feature development. Additionally, can we add the issues to the roadmap project - https://github.com/orgs/opensearch-project/projects/1

**Comment by anasalkouz** [comment]
> +1 to using the GitHub milestones feature.
> 
> Just a couple initial questions:
> 
> * Can anyone add an issue to a milestone?
It require the same access level as Labels, any one with Triage access or can apply milestones
Labels
![Screen Shot 2021-09-09 at 9 07 51 AM](https://user-images.githubusercontent.com/25317259/132722444-0a33837e-d82e-442a-9733-514e8640b9ae.png)
Milestones
![Screen Shot 2021-09-09 at 9 07 23 AM](https://user-images.githubusercontent.com/25317259/132722466-0684c1dd-62d2-46c7-8e40-8319cde1c00e.png)

> * Will milestones be time based (e.g., minor version milestone planned every N weeks)?
Yes, you can define due to date and track the progress based on the progress bar.
![Screen Shot 2021-09-09 at 8 46 55 AM](https://user-images.githubusercontent.com/25317259/132722692-b6c701e7-e74c-46b7-b862-909f6aa7ce64.png)

> * How do version labels cause explosion if maintainers are diligent in labeling and communicating with /asking questions of other contributors & community? Maybe combinatorial explosion of labels is a separate issue?
We will end adding ~8 labels every year to track minor versions.

> * Do we open a separate issue to discuss logically organizing source "components" into separate GitHub teams (e.g., analytics, search, distribution) to make notifications easier? And maybe how maintainers join those teams?
Thats sounds a good idea, will track it separately.

**Comment by anasalkouz** [comment]
> Milestones can be used to track progress of feature development. Additionally, can we add the issues to the roadmap project - https://github.com/orgs/opensearch-project/projects/1
Yes, nothing changed in the issue level, you can add issues to any project. Our proposal to tag issues for specific release using Milestones instead of labels.

**Comment by stockholmux** [comment]
I've not personally worked with milestones. @anasalkouz Do you have a few examples of projects using milestones well? I'd love to see how this plays out in practice.

Maybe folks in the community have strong opinions? Have we asked them?

**Comment by anasalkouz** [comment]
@stockholmux, here is an example of github project using milestones.
https://github.com/apache/rocketmq/milestones?direction=asc&sort=due_date
